spec/benchmarks/caching.rb

```diff
@@ -0,0 +1,63 @@
+# Usage: ruby spec/benchmarks/caching.rb
+
+# no replacement + many methods _can_ be faster than replacement + many
+# methods, but this depends on the values of METHOD_COUNT, OBJECT_COUNT,
+# and CALL_COUNT, so tweak them for your own scenario!
+
+require 'benchmark'
+require_relative '../../lib/batch_loader'
+
+METHOD_COUNT = 1000 # methods on the object with a large interface
+OBJECT_COUNT = 1000 # objects in the batch
+CALL_COUNT = 1000 # times a method is called on the loaded object
```

Comment:
you could potentially use https://github.com/evanphx/benchmark-ips instead, which is added as a development dependency. I like how it outputs the results with `Nx slower` :)

Reply:
nvm, from your comment descrition it might be a good idea to allow people to tweak the value manually 👍

Reply:
I think we could change it to use https://github.com/evanphx/benchmark-ips and remove the `CALL_COUNT` constant, maybe? Up to you!